src/AdsImporter/Importer.php

```diff
@@ -0,0 +1,245 @@
+        LoggerInterface $logger,
+        int $genesisTime,
+        int $blockSeqTime
+    ) {
+        $this->client = $client;
+        $this->databaseMigration = $databaseMigration;
+        $this->logger = $logger;
+        $this->genesisTime = $genesisTime;
+        $this->blockSeqTime = $blockSeqTime;
+        $this->importerResult = new ImporterResult();
+    }
+
+    public function import(): ImporterResult
+    {
+        $getMeResponse = $this->client->getMe();
+        $startTime = $this->getStartTime();
+        $endTime = (int)$getMeResponse->getPreviousBlockTime()->format('U');
+
+        $blockId = NumericalTransformation::decToHex($startTime);
+
+        // update nodes
+        $this->updateNodes();
+
+        do {
+            try {
+                $blockResponse = $this->client->getBlock($blockId);
+                /** @var Block $block */
+                $block = $blockResponse->getBlock();
+
+                $blockTransactions = $this->addPackagesFromBlock($block);
+
+                $block->setTransactionCount($blockTransactions);
+                $this->databaseMigration->addBlock($block);
+                ++$this->importerResult->blocks;
+            } catch (CommandException $ex) {
+                if ($ex->getCode() !== CommandError::GET_BLOCK_INFO_UNAVAILABLE) {
+                    $this->addExceptionToLog($ex, 'get_block', ['block' => $blockId]);
+                }
+            }
+
+
+            $startTime += $this->blockSeqTime;
+            $blockId = NumericalTransformation::decToHex($startTime);
+        } while ($startTime <= $endTime);
+
+        return $this->importerResult;
+    }
+
+    private function getStartTime(): int
+    {
+        $from = $this->databaseMigration->getNewestBlockTime();
+
+        if ($from) {
+            return $from + $this->blockSeqTime;
+        }
+
+        return $this->genesisTime;
+    }
+
+    private function updateNodes(): void
+    {
+        try {
+            $blockResponse = $this->client->getBlock();
+        } catch (CommandException $ex) {
+            throw new AdsClientException('Cannot proceed importing data');
+        }
+
+        $nodes = $blockResponse->getBlock()->getNodes();
+
+        /** @var Node $node */
+        foreach ($nodes as $node) {
+            if ($node->getId() === '0000') { // special node
+                continue;
+            }
+
+            $this->updateAccounts($node);
+
+            $this->databaseMigration->addOrUpdateNode($node);
+            ++$this->importerResult->nodes;
+        }
+    }
+
+    /**
+     * @param Node $node
+     *
+     * todo how we can generate hash (or something) of a node to not update anything if there is no changes
+     */
+    private function updateAccounts(Node $node): void
+    {
+        $accountResponse = $this->client->getAccounts($node->getId());
+        $accounts = $accountResponse->getAccounts(); // todo check if `getAccounts` method always returns array
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1135)